frontend/public/components/routes.tsx

```diff
@@ -200,26 +203,48 @@ const RouteListRow: React.SFC<RoutesRowProps> = ({obj: route}) => <ResourceRow o
   <div className="col-lg-2 hidden-md hidden-sm hidden-xs"><RouteStatus obj={route} /></div>
 </ResourceRow>;
 
-const TLSSettings = props => <span>
-  {!props.tls && 'TLS is not enabled.'}
-  {props.tls && <dl>
-    <dt>Termination Type</dt>
-    <dd>{props.tls.termination}</dd>
-    <dt>Insecure Traffic</dt>
-    <dd>{props.tls.insecureEdgeTerminationPolicy || '-'}</dd>
-    <dt>Certificate</dt>
-    <dd>{props.tls.certificate || '-'}</dd>
-    <dt>Key</dt>
-    <dd>{props.tls.key || '-'}</dd>
-    <dt>CA Certificate</dt>
-    <dd>{props.tls.caCertificate || '-'}</dd>
-    {_.get(props.tls, 'termination') === 'reencrypt' && <span>
-      <dt>Destination CA Cert</dt>
-      <dd>{props.tls.destinationCACertificate || '-'}</dd>
-    </span>}
-  </dl>
+class TLSSettings extends SafetyFirst<TLSDataProps, TLSDataState> {
+  constructor(props) {
+    super(props);
+    this.state = { showPrivateKey: false };
+    this.toggleKey = this.toggleKey.bind(this);
+  }
+
+  toggleKey() {
+    this.setState({ showPrivateKey: !this.state.showPrivateKey });
+  }
+
+  render() {
+    const { showPrivateKey } = this.state;
+    const visibleKeyValue = showPrivateKey ? this.props.tls.key : <MaskedData /> ;
+
+    return !this.props.tls ? 'TLS is not enabled.'
+      : <dl>
+        <dt>Termination Type</dt>
+        <dd>{this.props.tls.termination}</dd>
+        <dt>Insecure Traffic</dt>
+        <dd>{this.props.tls.insecureEdgeTerminationPolicy || '-'}</dd>
+        <dt>Certificate</dt>
+        <dd>{this.props.tls.certificate ? <CopyToClipboard value={this.props.tls.certificate}/> : '-'}</dd>
+        <dt>Key {this.props.tls.key &&
+        <button className="btn btn-link co-mask-data__btn" type="button" onClick={this.toggleKey}>
```

Comment:
Do we want the button right-aligned like the secret page?

Reply:
Seems like a @openshift/team-ux-review question :-)  My only concern is that perhaps users would think they would have to 'Reveal' the value in order to 'Copy' it  -if the buttons are close together.